akka-actor/src/main/scala/akka/io/dns/AsyncDnsManager.scala

```diff
@@ -0,0 +1,69 @@
+/*
+ * Copyright (C) 2018 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.io.dns
+
+import java.util.concurrent.TimeUnit
+
+import akka.actor.{ Actor, ActorLogging, Deploy, Props }
+import akka.dispatch.{ RequiresMessageQueue, UnboundedMessageQueueSemantics }
+import akka.io.{ Dns, DnsExt, PeriodicCacheCleanup }
+import akka.routing.FromConfig
+
+import scala.concurrent.duration.Duration
+
+final class AsyncDnsManager(val ext: DnsExt) extends Actor with RequiresMessageQueue[UnboundedMessageQueueSemantics] with ActorLogging {
+  import AsyncDnsManager._
+  implicit val ec = context.dispatcher
+
+  private var oldProtocolWarningLoggedTimes = 0
```

Comment:
to not keep logging forever if people make that mistake -- could be not their fault for example